jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -179,6 +179,15 @@ def _on_glue_y_display_unit_changed(self, y_unit_str):
                 self.flux_unit.selected,
                 self.angle_unit.selected
             )
+            self.hub.broadcast(GlobalDisplayUnitChanged("sb", self.sb_unit_selected, sender=self))
+
+        else:
```

Comment:
I don't quite understand how this and the `if` statement above are working to distinguis between surface brightness and flux - is `if self.app.data_collection[0]` not just checking to see if there's _anything_ in the data collection? I don't think you wrote the original case, but do you have a better understanding of what's going on here?

Reply:
youre right this is in the wrong place, once sec